File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/models/dto/BurguerDTO.java
package Ficheros.BurguerPig.models.dto;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.io.Serializable;
import java.util.List;

@Root(name = "hamburguesa") // Para XML
public class BurguerDTO implements Serializable {
    @Attribute(name = "name")
    private String name;

    // El inline es para que no se vea la jerarquía de la lista, y sea más legible
    @ElementList(name = "ingredientes", inline = true)
    private List<IngredienteDTO> ingredients;

    @Attribute(name = "uuid")
    private String uuid;

    @Attribute(name = "precio")
    private String price;

    public BurguerDTO() {
        // Constructor vacío sin argumentos
    }

    public BurguerDTO(
            String name,
            List<IngredienteDTO> ingredients,
            String uuid,
            String price) {
        this.name = name;
        this.ingredients = ingredients;
        this.uuid = uuid;
        this.price = price;
    }

    public String getName() {
        return name;
    }


    public List<IngredienteDTO> getIngredients() {
        return ingredients;
    }


    public String getUuid() {
        return uuid;
    }


    public String getPrice() {
        return price;
    }

}

File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Kotlin/src/main/kotlin/Ficheros/BurguerPig/models/dto/BurguerDTO.kt
package Ficheros.BurguerPig.models.dto

import com.squareup.moshi.JsonAdapter
import org.simpleframework.xml.Attribute
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "hamburguesa")
data class BurguerDTO(
    @field:Attribute(name = "name")
    @param:Attribute(name = "name")
    val name: String,
    // El inline es para que no se vea la jerarquía de la lista, y sea más legible
    @field:ElementList(name = "ingredientes", inline = true)
    @param:ElementList(name = "ingredientes", inline = true)
    val ingredients: List<IngredientDTO>,
    @field:Attribute(name = "uuid")
    @param:Attribute(name = "uuid")
    val uuid: String,
    @field:Attribute(name = "precio")
    @param:Attribute(name = "precio")
    val price: String
)

@Root(name = "hamburguesas")
data class BurguerListDto(
    // El inline es para que no se vea la jerarquía de la lista, y sea más legible
    @field:ElementList(name = "hamburguesa", inline = true)
    @param:ElementList(name = "hamburguesa", inline = true)
    val burguers: List<BurguerDTO>
)